### ISSUE TITLE
Associate data source from folder

### ISSUE THREAD

**Issue by mittalyashu** [dbeaver/dbeaver#9388]
**Is your feature request related to a problem? Please describe.**

Is difficult to associate data source to each filter or by selecting multiple files manually and there can be multiple sql script files in sub-directory.

**Describe the solution you'd like**

Option to associate data source to the folder and its sub files and folder.

**Comment by kseniiaguzeeva** [comment]
The similar feature already exists (associate data source for SQL Script). For this:
1. Go to Projects>Scripts folder
2. Select Associate data source in a script context menu.

Or please describe with more details how you would like the feature looks like.

**Comment by mittalyashu** [comment]
Yes, I am aware of that option, but I am looking to attach the whole folder (sub files and sub folder) to associate data source at once, instead of doing it file-by-file.

**Comment by kseniiaguzeeva** [comment]
Thank you for clarifying

**Comment by mittalyashu** [comment]
I am not sure if this feature is requested or work in progress but if the file is associated with any source we don't even know just by looking at the icon.

With the unique icon for associated file it will help in understanding if the file is linked or not

And we don't have to open it again and again to check it.

**Comment by budlin2** [comment]
Doesn't look like any headway has been made on this, but I think this would be a useful feature. This way you could right click on the folder, Create > Create New File, and quickly have a script in the location you want associated with the data source you want.

**Comment by E1izabeth** [comment]
Linked with #8028

**Comment by Karmey** [comment]
Hello. In my work, I use several connections to different databases and different schemas, while a lot of folds with sql editors are open. It would be great to be able to filter tabs based on connection or have a combobox at the top as a filter so you can filter tabs with editors

**Comment by polczak-itt** [comment]
Very surprised there aren't many votes for this. I'm only using DBeaver for a couple of weeks now and I very quickly found it unhelpful that there's no better way for this than manually selecting newly added files (with Ctrl or Shift) each time some new script gets added to the folder...

**Comment by lukasheinz92** [comment]
Is there ETA for implementing this feature? It would be really helpfull for me.

**Comment by E1izabeth** [comment]
It's not planned to any milestone for now.
